pkg/plugins/identity/oidc/provider.go

```diff
@@ -0,0 +1,261 @@
+limitations under the License.
+*/
+
+package oidc
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"github.com/fidelity/kconnect/pkg/config"
+	"github.com/fidelity/kconnect/pkg/oidc"
+	"github.com/fidelity/kconnect/pkg/provider"
+	"github.com/fidelity/kconnect/pkg/provider/identity"
+	"github.com/fidelity/kconnect/pkg/provider/registry"
+
+	kconnectv1alpha "github.com/fidelity/kconnect/api/v1alpha1"
+	khttp "github.com/fidelity/kconnect/pkg/http"
+)
+
+const (
+	ProviderName      = "oidc"
+	True              = "true"
+	FailCallConfigUrl = "Failed accessing config url, you need to continue to input missing information."
+)
+
+func init() {
+	if err := registry.RegisterIdentityPlugin(&registry.IdentityPluginRegistration{
+		PluginRegistration: registry.PluginRegistration{
+			Name:                   ProviderName,
+			UsageExample:           "",
+			ConfigurationItemsFunc: ConfigurationItems,
+		},
+		CreateFunc: New,
+	}); err != nil {
+		zap.S().Fatalw("Failed to register OIDC identity plugin", "error", err)
+	}
+}
+
+// New will create a new OIDC identity provider
+func New(input *provider.PluginCreationInput) (identity.Provider, error) {
+	return &oidcIdentityProvider{
+		logger:      input.Logger,
+		interactive: input.IsInteractice,
+	}, nil
+}
+
+type oidcIdentityProvider struct {
+	logger      *zap.SugaredLogger
+	interactive bool
+}
+
+type providerConfig struct {
+	OidcServer string `json:"oidc-server"`
+	OidcId     string `json:"oidc-client-id"`
+	OidcSecret string `json:"oidc-client-secret"`
+	UsePkce    string `json:"oidc-use-pkce"`
+}
+
+func (p *oidcIdentityProvider) Name() string {
+	return ProviderName
+}
+
+// Authenticate will generate authentication config.
+func (p *oidcIdentityProvider) Authenticate(ctx context.Context, input *identity.AuthenticateInput) (*identity.AuthenticateOutput, error) {
+	p.logger.Info("using oidc for authentication")
+
+	p.getConfigFromUrl(input.ConfigSet)
+
+	cfg := &providerConfig{}
+	if err := config.Unmarshall(input.ConfigSet, cfg); err != nil {
+		return nil, fmt.Errorf("unmarshalling config into providerConfig: %w", err)
+	}
+
+	id := &oidc.Identity{
+		OidcServer: cfg.OidcServer,
+		OidcId:     cfg.OidcId,
+		OidcSecret: cfg.OidcSecret,
+		UsePkce:    cfg.UsePkce,
+	}
+
+	ids, err := p.readRequiredFields(*id)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = executeOidcLogin(ids); err != nil {
+		return nil, err
+	}
+
+	return &identity.AuthenticateOutput{
+		Identity: &ids,
+	}, nil
+}
+
+func executeOidcLogin(id oidc.Identity) error {
+
+	args := []string{
+		"oidc-login",
+		"get-token",
+		"--oidc-issuer-url=" + id.OidcServer,
+		"--oidc-client-id=" + id.OidcId,
+	}
+
+	if id.UsePkce == True {
+		args = append(args, "--oidc-use-pkce")
+	} else {
+		args = append(args, "--oidc-client-secret="+id.OidcSecret)
+	}
+	args = append(args, "--insecure-skip-tls-verify")
```

Comment:
same thing here about, do we really need to have this as the default behavior?

Reply:
Same as above, added flag, to not make this as default behavior. Thanks!